core/templates/components/summary-tile/story-summary-tile.component.spec.ts

```diff
@@ -0,0 +1,448 @@
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1', 'node2', 'node3', 'node4', 'node5'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    // Will show view all button as number of nodes (5) is more than chapters
+    // displayed (3).
+    expect(component.showButton).toBe(true);
+  });
+
+  it('should show user\'s progress', () => {
+    // StorySummary with 1 node with 0 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: [],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+    let circumference = (2 * 20 * Math.PI).toString();
+
+    component.ngOnInit();
+
+    expect(component.strokeDashArrayValues).toBe('');
+    expect(component.completedStrokeDashArrayValues).toBe('0 ' + circumference);
+
+    // StorySummary with 1 node with 1 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    expect(component.strokeDashArrayValues).toBe('');
+    expect(component.completedStrokeDashArrayValues).toBe('');
+
+    // StorySummary with 5 node with 3 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1', 'node2', 'node3', 'node4', 'node5'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1', 'node2', 'node3'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    // The values are calculated by the same formula as explained above.
+    expect(component.strokeDashArrayValues).toBe('20.132741228718345 5');
+    expect(component.completedStrokeDashArrayValues).toBe(
+      '20.132741228718345 5 20.132741228718345 5' +
+      ' 20.132741228718345 55.26548245743669');
+  });
+
+  it('should link chapter title to URL', () => {
```

Comment:
`should return the chapter URL when the chapter title is provided`

Reply:
Done